### PR TITLE
Selenium tests for Duplicate Check

### PR DESCRIPTION
**Pull request by PenghaiZhang**
* Test Editbox duplicate check
* Test attachments(files and links) duplicate check
* Test Duplicate data page
* Add a new collection to institution fiveo
* Make attachment warning consistent with Editbox warning

### CODE REVIEW

File: autotest/OldTests/src/test/java/com/tle/webtests/test/contribute/UniquenessTest.java [hunk 1]
```diff
@@ -51,4 +73,249 @@ public void contributeSecond() {
...
+  @Test
+  public void createDuplicateCheckTestData() {
+    ContributePage contributePage = new ContributePage(context).load();
+    WizardPageTab wizardPage = contributePage.openWizard(DUPLICATE_CHECK_COLLECTION);
+    wizardPage.editbox(1, DUPLICATE_CHECK_ITEM_NAME);
+    wizardPage.editbox(2, DUPLICATE_CHECK_ITEM_ID);
+    wizardPage.addFile(3, FIRST_FILE_NAME);
+    wizardPage.addUrl(3, FIRST_LINK_NAME, FIRST_LINK_NAME);
+    wizardPage.save().publish();
+  }
```
Comment (edalex-ian) [inline autotest/OldTests/src/test/java/com/tle/webtests/test/contribute/UniquenessTest.java]: Is this just setting up data? If so, it doesn't need `@Test` and should go in a `setUp()` method.

File: autotest/OldTests/src/test/java/com/tle/webtests/test/contribute/UniquenessTest.java [hunk 2]
```diff
@@ -51,4 +73,249 @@ public void contributeSecond() {
...
+    confirmationDialog.cancel(wizardPage);
+    warningDisplayed = findWarningLink(wizardPage, controlDivId, true);
+    assertTrue(warningDisplayed);
+    checkDuplicateItem(true, DUPLICATE_CHECK_ITEM_ID);
+    wizardPage.get();
+    wizardPage.editbox(2, DUPLICATE_CHECK_ITEM_DIFFERENT_ID);
+    wizardPage.save().cancel(wizardPage);
+    warningDisplayed = findWarningLink(wizardPage, controlDivId, false);
+    assertFalse(warningDisplayed);
+  }
```
Comment (edalex-ian) [inline autotest/OldTests/src/test/java/com/tle/webtests/test/contribute/UniquenessTest.java]: This looks like (and highlighted by your comments) two tests in one. So why not done as two standalone tests?

File: autotest/OldTests/src/test/java/com/tle/webtests/test/contribute/UniquenessTest.java [hunk 3]
```diff
@@ -51,4 +73,249 @@ public void contributeSecond() {
...
+          duplicateContent.getText(), "'" + duplicateContentText + "' is also used by:");
+    }
+
+    By duplicatesFoundXpath = By.xpath("//a[text()='" + DUPLICATE_CHECK_ITEM_NAME + "']");
+    WebElement duplicateItem =
+        duplicateDataPage
+            .getWaiter()
+            .until(ExpectedConditions.visibilityOfElementLocated(duplicatesFoundXpath));
+    // Check if the correct duplicate item is found
+    Assert.assertEquals(duplicateItem.getText(), "uniquenessTest - Duplicate Check");
```
Comment (edalex-ian) [inline autotest/OldTests/src/test/java/com/tle/webtests/test/contribute/UniquenessTest.java]: You've got a constant for that String don't you?

```suggestion
    Assert.assertEquals(duplicateItem.getText(), DUPLICATE_CHECK_ITEM_NAME);
```

File: autotest/OldTests/src/test/java/com/tle/webtests/test/contribute/UniquenessTest.java [hunk 4]
```diff
@@ -51,4 +73,249 @@ public void contributeSecond() {
...
+          By.xpath(
+              "//div[starts-with(@class,'duplicate')]//*[contains(text(),"
+                  + quoteXPath(duplicateContentText)
+                  + ")]");
+      WebElement duplicateContent =
+          duplicateDataPage
+              .getWaiter()
+              .until(ExpectedConditions.visibilityOfElementLocated(duplicateContentXpath));
+      Assert.assertEquals(
+          duplicateContent.getText(), "'" + duplicateContentText + "' is also used by:");
```
Comment (edalex-ian) [inline autotest/OldTests/src/test/java/com/tle/webtests/test/contribute/UniquenessTest.java]: I wonder if for maintenance sake, you should use the language strings for that second param. :thinking: 

(I also wonder if you can use them from this context.)

Comment (PenghaiZhang) [reply autotest/OldTests/src/test/java/com/tle/webtests/test/contribute/UniquenessTest.java]: Hi Ian, can I leave this question until tomorrow?  I couldn't find any usage of language strings in these tests and Charlie doesn't know about this, so I want to seek some help from Aaron.

**Comment by PenghaiZhang** [comment]
> I reckon good to go.
> 
> Did you confirm that the UniquenessTest is executed as part of the current Travis suites?

Yes, UniquenessTest is part of `Contribution`

**Pull request merged**